### PR TITLE
wip: add new make rule for checking if module/probe are available for the system

### PR DESCRIPTION
**Pull request by Issif**
This PR adds a new make rule `check-driver` for allowing users to check if drivers are available for their system or their kubernetes cluster.

The rule takes 2 arguments:
```
make check-driver -e CHECK_RUNTIME=kubernetes -e FALCO_VERSION=0.29.1
```
With:
- `FALCO_VERSION`: version of Falco you want to check for your system, only for `check` (default is last Falco version)
- `CHECK_RUNTIME`: runtime for performing `check`: locale (default), docker, kubernetes
- 
```shell
make check-driver

• Falco version: 0.30.0
• Lib version: 3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4
• Probe: x
• Module: x

We're sorry, the ebpf probe or kernel module is not already built for your system.

Please, help the community and yourself by submitting a PR on https://github.com/falcosecurity/test-infra
for adding the following content in new file 'driverkit/config/3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4/manjaro_5.15.12-1-MANJARO_1.yaml':

kernelversion: 1
kernelrelease: 5.15.12-1-MANJARO
target: manjaro
output:
  module: output/3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4/falco_manjaro_5.15.12-1-MANJARO_1.ko
  probe: output/3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4/falco_manjaro_5.15.12-1-MANJARO_1.o

You can also run 'make generate -e TARGET_DISTRO=manjaro -e TARGET_KERNEL=5.15.12-1-MANJARO_1'
from https://github.com/falcosecurity/test-infra/driverkit, see documentation: https://github.com/falcosecuirty/test-infra/tree/master/driverkit#faq
it will create .yaml files for your system for all available versions of Falco libs.
```
```shell
make check-driver -e FALCO_VERSION=0.29.1

• Falco version: 0.29.1
• Lib version: 17f5df52a7d9ed6bb12d3b1768460def8439936d
• Probe: x
• Module: x

We're sorry, the ebpf probe or kernel module is not already built for your system.

Please, help the community and yourself by submitting a PR on https://github.com/falcosecurity/test-infra
for adding the following content in new file 'driverkit/config/17f5df52a7d9ed6bb12d3b1768460def8439936d/manjaro_5.15.12-1-MANJARO_1.yaml':

kernelversion: 1
kernelrelease: 5.15.12-1-MANJARO
target: manjaro
output:
  module: output/17f5df52a7d9ed6bb12d3b1768460def8439936d/falco_manjaro_5.15.12-1-MANJARO_1.ko
  probe: output/17f5df52a7d9ed6bb12d3b1768460def8439936d/falco_manjaro_5.15.12-1-MANJARO_1.o

You can also run 'make generate -e TARGET_DISTRO=manjaro -e TARGET_KERNEL=5.15.12-1-MANJARO_1'
from https://github.com/falcosecurity/test-infra/driverkit, see documentation: https://github.com/falcosecuirty/test-infra/tree/master/driverkit#faq
it will create .yaml files for your system for all available versions of Falco libs.
```

```shell
make check-driver

• Falco version: 0.30.0
• Lib version: 3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4
• Probe: x
• Module: ✔

Congratulations! Your system is ready for running Falco!

• Get Started in Falco.org (https://falco.org)
• Check out the Falco project and contribute in Github (https://github.com/falcosecurity/falco)
• Get involved in the Falco community (https://github.com/falcosecurity/community)
• Meet the maintainers on the Falco Slack (https://kubernetes.slack.com#falco)
• Follow @falco_org on Twitter
```

```shell
make check-driver -e CHECK_RUNTIME=kubernetes                                                                                                                                          

job.batch/falco-driver-check created
job.batch/falco-driver-check condition met

• Falco version: 0.30.0
• Lib version: 3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4
• Probe: x
• Module: x

We're sorry, the ebpf probe or kernel module is not already built for your system.

Please, help the community and yourself by submitting a PR on https://github.com/falcosecurity/test-infra
for adding the following content in new file 'driverkit/config/3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4/falco_alpine_5.10.82-0-virt_1.yaml':

kernelversion: 1
kernelrelease: falco_alpine_5.10.82-0-virt_1
target: alpine
output:
  module: output/3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4/falco_alpine_5.10.82-0-virt_1.ko
  probe: output/3aa7a83bf7b9e6229a3824e3fd1f4452d1e95cb4/falco_alpine_5.10.82-0-virt_1.o

You can also run 'make generate -e TARGET_DISTRO=alpine -e TARGET_KERNEL=falco_alpine_5.10.82-0-virt_1'
from https://github.com/falcosecurity/test-infra/driverkit, see documentation: https://github.com/falcosecuirty/test-infra/tree/master/driverkit#faq
it will create .yaml files for your system for all available versions of Falco libs.

job.batch "falco-driver-check" deleted
```

This is a backport of https://github.com/Issif/falco-driver-check

### CODE REVIEW

File: images/check-driver/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+IMAGE = falcosecurity/falco-driver-check
```
Comment (leogr) [inline images/check-driver/Makefile:1]: I guess this is intended to be published to the docker.io registry, right?
If so:
- please, use the FQN
- we have to create the repo in dockerhub
- we have to handle the auth (AFAIK, pushing to dockerhub is not yet implemented in test-infra)

Comment (leogr) [reply images/check-driver/Makefile:1]: Furthermore, there's no job configured to automatically build and publish the image when a code change gets merged. We should do that in a similar way we already do for prow images.

File: driverkit/misc/check-driver-job.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+apiVersion: batch/v1
```
Comment (leogr) [inline driverkit/misc/check-driver-job.yaml:1]: I could not find any link to this file. I guess it is intended as an example for the users, isn't it?

If yes, could we document it?

**Review by leogr — COMMENTED** [review]
Hey Thomas,

Thank you for this feature!

I have just found some issue (see comments below).

File: images/check-driver/Makefile (hunk 1 above)
Comment (Issif) [reply images/check-driver/Makefile]: You're right, and I might need some help for adding this with Prow. I fixed the FQDN as requested.

File: images/check-driver/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+curl -s https://raw.githubusercontent.com/Issif/test-infra/master/driverkit/utils/checkdriver -o checkdriver.sh
```
[the remark this answers is not in the document]
Comment (Issif) [reply images/check-driver/entrypoint.sh]: fixed

File: driverkit/misc/check-driver-job.yaml (hunk 2 above)
Comment (Issif) [reply driverkit/misc/check-driver-job.yaml:1]: This  yaml is used when you specify `CHECK_RUNTIME=kubernetes`, this creates a Job with the Docker image you noticed. This is transparent for users.

**Comment by maxgio92** [comment]
As discussed, we'll wait for related artifacts to be published. Maybe we can put this PR in WIP status @Issif? WDYT?

**Pull request closed without merging**